lucene/core/src/java/org/apache/lucene/util/bkd/BKDRadixSelector.java

```diff
@@ -294,18 +335,37 @@ protected int byteAt(int i, int k) {
       }
     }.select(from, to, partitionPoint);
 
-    for (int i = from; i < to; i++) {
-      points.getPackedValueSlice(i, bytesRef1);
-      int docID = points.docIDs[i];
-      if (i < partitionPoint) {
-        left.append(bytesRef1, docID);
-      } else {
-        right.append(bytesRef1, docID);
-      }
-    }
     byte[] partition = new byte[bytesPerDim];
     points.getPackedValueSlice(partitionPoint, bytesRef1);
     System.arraycopy(bytesRef1.bytes, bytesRef1.offset + dim * bytesPerDim, partition, 0, bytesPerDim);
     return partition;
   }
+
+  PointWriter getPointWriter(long count, String desc) throws IOException {
+    if (count <= maxPointsSortInHeap / 2) {
```

Comment:
why do we divide by 2? can you add a comment?

Reply:
We divided by 2 because as we recurse, we have two `heapPointWriter` as maximum, so they should not hold more than half of the `maxPointsSortInHeap`.